content/output/demographic/_index.Rmd

```diff
@@ -0,0 +1,98 @@
+---
+title: "Create Demographic table"
+author: "Tadeusz Lewandowski"
+date: "16/08/2022"
+output: html_document
+description: "An example of how to generate an Demogrphic output based on ADaM dataset"
+---
+
+## Introduction
+
+This guide will show you how two pharmaverse packages, along with some from tidyverse, can be used to create a Demographic table as an end-to-end, using the Admiral ADSL data as input.
+
+The two packages used with a brief description of their purpose are as follows:
+
+* [`{rtables}`](https://roche.github.io/rtables/index.html): designed to create and display complex tables with R.
+* [`{tern}`](https://insightsengineering.github.io/tern/main/index.html): contains analysis functions to create tables and graphs used for clinical trial reporting.
+
+It is important to understand [`{rtables}`](https://roche.github.io/rtables/index.html) grammar by reading through the above linked package site, as these are fundamental to being able to use [`{tern}`](https://insightsengineering.github.io/tern/main/index.html). 
+
+## Load Data and Required pharmaverse Packages
+
+After installation of packages, the first step is to load our pharmaverse packages and input data.
+
+```{r setup, message=FALSE, warning=FALSE, results='hold'}
+options(repos = c(
+  pharmaverse = 'https://pharmaverse.r-universe.dev',
+  CRAN = 'https://cloud.r-project.org'))
+
+library(admiral)
+library(rtables)
+library(tern)
+library(dplyr)
+
+# Read in input ADaM data 
+data("admiral_adsl")
+
+```
+  
+## Start preprocessing
+
+The first step we are going to encode missing entries across groups of categorical variables in a data frame `adsl`.
+
+
+```{r, error=TRUE}
+adsl <- df_explicit_na(admiral_adsl)
+```
+
+Now we will change some and add new variable `AGEGRP`. In the new variable we also are going to add levels in order to keep the required sequence of new categories on the final output.
+
+```{r}
+adsl <- adsl %>%
+  mutate(
+    SEX = factor(case_when(
+      SEX == "M" ~ "Male",
+      SEX == "F" ~ "Female",
+      SEX == "U" ~ "Unknown",
+      SEX == "UNDIFFERENTIATED" ~ "Undifferentiated"
+    )),
+    AGEGRP = factor(case_when(
```

Comment:
we already have AGEGR1 derived in the ADSL example, so could we use this directly? re-formatting variables for display is fine but i wouldn't want people to think we advocate derivations in TLG programs, as these would be done in ADaM

Reply:
Ok good point, I can amend it, but will leave some code to present that the pre-processing could be done at the output level program. As the same example I have for the Shin, so I will synchronize booth codes.